apps/api/src/server/app/state-machine/household-appeal.machine.js

```diff
@@ -3,31 +3,59 @@ import { validationStates, validationActions } from './validation-states.js';
 import { lpaQuestionnaireStates, lpaQuestionnaireActions } from './lpa-questionnaire-states.js';
 import { inspectorStates, inspectorActions } from './inspector-states.js';
 
-const createHouseholpAppealMachine = function(context) {
-	return createMachine({
-		id: 'household_appeal',
-		context: context,
-		initial: 'received_appeal',
-		states: {
-			...validationStates,
-			...lpaQuestionnaireStates,
-			...inspectorStates
+/**
+ * @typedef {object} TransitionStateErrorConfig
+ * @property {string} type - The state machine action being performed.
+ * @property {object} context - The context on which the state machine error occurred.
+ */
+
+export class TransitionStateError extends Error {
+	/**
+	 * @param {string} message
+	 * @param {TransitionStateErrorConfig} config
+	 */
+	constructor(message, { action, appeal }) {
+		super(message);
+		this.type = action;
+		this.context = appeal;
+	}
+}
+
+const createHouseholpAppealMachine = function (context) {
+	return createMachine(
+		{
+			id: 'household_appeal',
+			context: context,
+			initial: 'received_appeal',
+			states: {
+				...validationStates,
+				...lpaQuestionnaireStates,
+				...inspectorStates
+			}
 		},
-	}, {
-		actions: {
-			...validationActions,
-			...lpaQuestionnaireActions,
-			...inspectorActions
+		{
+			actions: {
+				...validationActions,
+				...lpaQuestionnaireActions,
+				...inspectorActions
+			}
 		}
-	});
+	);
 };
 
-const transitionState = function(context, status, machineAction) {
+const transitionState = function (context, status, machineAction) {
 	const service = interpret(createHouseholpAppealMachine(context));
 	service.start(status);
 	service.send({ type: machineAction });
 	const nextState = service.state;
 	service.stop();
+
+	if (nextState.value === status) {
```

Comment:
Not sure that this is the best way to check if there has been a transition. What if we have an instance when the action loops us back to the initial state of the appeal?

Reply:
There's a use case where we could "advance" a state to itself? Mmmm, ok... well, I don't know this state machine library at all – is there some way to check it did something valid?

Reply:
yep, we check if the transition was valid in the household appeal state machine tests. So we can check if the `nextState.changed` is their `false` or `undefined` and that would tell us that the transition did not go through

Reply:
Ok, let me change this to use `.changed`

Reply:
Added, and also for now, had to add an opt-in flag to the transition state throwing an error as it breaks some tests in validation